### PR TITLE
Updated Tomlyn dll with latest version (0.16.2)

### PR DESCRIPTION
**Pull request by Gilford92**
Updated Tomlyn dll with latest version (0.16.2)
Updates from Unity to the file "UpgradePackagesManager.cs" due to new api

### CODE REVIEW

**Comment by jespersmith** [comment]
Is there any technical reason to not prefix the DLL with a hash? I did that to avoid conflicts in the past, best to keep doing it if it doesn't break the the entities package.

File: Editor/Halodi/PackageRegistry/Core/UpgradePackagesManager.cs [hunk 1]
```diff
@@ -82,22 +82,22 @@ public PackageUpgradeState(UnityEditor.PackageManager.PackageInfo info)
...
                     }
                     
-                    hasVerified = !String.IsNullOrWhiteSpace(info.versions.verified);
+                    hasVerified = !String.IsNullOrWhiteSpace(info.versions.recommended);
                     if(hasVerified)
                     {
                         try
                         {
-                            verifiedVersion = SemVer.Parse(info.versions.verified);
+                            verifiedVersion = SemVer.Parse(info.versions.recommended);
```
Comment (jespersmith) [inline Editor/Halodi/PackageRegistry/Core/UpgradePackagesManager.cs:90]: Breaks Unity 2021/2022.

File: Editor/Halodi.PackageRegistryManager.Editor.asmdef [hunk 2]
```diff
@@ -11,9 +11,9 @@
     "allowUnsafeCode": false,
     "overrideReferences": true,
     "precompiledReferences": [
-        "860BB79D.Tomlyn.dll",
```
Comment (StephenHodgson) [inline Editor/Halodi.PackageRegistryManager.Editor.asmdef:14]: does changing the name also rerecate collision from #25 ?

File: Editor/Halodi/PackageRegistry/Core/UpgradePackagesManager.cs (hunk 1 above)
Comment (Gilford92) [reply Editor/Halodi/PackageRegistry/Core/UpgradePackagesManager.cs]: Added precompiled check to avoid issues with previous versions of Unity

**Comment by Gilford92** [comment]
> Is there any technical reason to not prefix the DLL with a hash? I did that to avoid conflicts in the past, best to keep doing it if it doesn't break the the entities package.

Unfortunately this is necessary to make the Entities package work. If I don't rename it the exception cited in the linked issue will be back and you cannot use Halodi with Entities package.

You can simply test it in this way:
- Create a new project with halodi package
- Rename Tomlyn.dll to 860BB79D.Tomlyn.dll
- Download the Entities package from Unity package manager
- You are gonna see the exception cited in the issue

I think this is happening because of the reply in this report
https://issuetracker.unity3d.com/issues/warnings-are-logged-for-editor-only-assemblies-when-building-project

Basically when there are "multiple assemblies with the same name in a project, we'll automatically pick up the highest version available in the project". So I think that because Tomlyn.dll in Entities package and the one in Halodi will have a different name, Unity tries to import both of them causing conflicts between these 2. But this is my supposition.

File: Editor/Halodi.PackageRegistryManager.Editor.asmdef (hunk 2 above)
Comment (Gilford92) [reply Editor/Halodi.PackageRegistryManager.Editor.asmdef:14]: No idea, but as said in the previous comment I need to make this change otherwise the Entities package will not work.

Is there a way to test if #16 is not a problem anymore?

**Comment by StephenHodgson** [comment]
Sounds like Unity needs to pull Tomlyn out into its own package like they did with newtonsoft

**Comment by jespersmith** [comment]
Thanks for the changes, I'll merge it. If Tomlyn still breaks, an old comment of mine would be the solution

> Not sure how to fix the Tomlyn issue. I can unpack Tomlyn and make an assembly reference instead, which would solve this warning. Not sure about the disadvantages.
